src/transform/src/literal_constraints.rs

```diff
@@ -155,7 +155,8 @@ impl LiteralConstraints {
                             // common here, so we explicitly add it.)
                             keys: vec![(0..key.len()).collect()],
                         },
-                    };
+                    }
+                    .arrange_by(&[(0..key.len()).map(MirScalarExpr::Column).collect_vec()]);
```

Comment:
Is this something `ArrangeBy` something that at the moment would be added by `JoinImplementation` or some other transform?

Reply:
`JoinImplementation` is not putting the `ArrangeBy`, because [it explicitly skips touching `IndexedFilter` joins](https://github.com/MaterializeInc/materialize/blob/24c09f273de612f8f6653bc40d28b88799039ba6/src/transform/src/join_implementation.rs#L143). And also nothing else was putting an `ArrangeBy` here, so the added `soft_panic` was failing for `IndexedFilter` joins before I added this `ArrangeBy`.

Reply:
OK, then I would add this PR to the list of fixes in your comment after the panic.